pkg/scuzz/ata.go

```diff
@@ -78,11 +84,81 @@ type (
 	// for all kernels what the rules are.
 	dataBlock [oldSchoolBlockLen]byte
 
+	// Some ATA blocks are best dealt with as blocks of "words".
+	// They are in big-endian order.
+	wordBlock [oldSchoolBlockLen / 2]uint16
+
 	// commandDataBlock defines a command and any associated data.
 	commandDataBlock [ata16Len]byte
 
 	// statusBlock is the status returned from a drive operation.
 	statusBlock [maxStatusBlockLen]byte
 
 	direction int32
+
+	// ataString is a 10 words, bytes swapped.
+	ataString [10]uint16
 )
+
+func (b dataBlock) toWordBlock() (wordBlock, error) {
+	var w wordBlock
+	err := binary.Read(bytes.NewBuffer(b[:]), binary.BigEndian, &w)
+	return w, err
+}
+
+// mustLBA confirms that we are dealing with an LBA device.
+// This means a post-2003 device. The standard hdparm command deals
+// with all kinds of obsolete stuff; we don't care.
+// The spec does not have a name for the bits or the offsets
+// other than "Obsolete", "Retired", "Must be zero" or "Must be one".
+// We follow the practice of existing code of not naming them either.
+func (w wordBlock) mustLBA() error {
+	var check = []struct {
+		off  int
+		mask uint16
+		bit  uint16
+	}{
+		{0, 0x8000, 0x8000},
+		{49, 0x200, 0x200},
+		{83, 0xc000, 0x4000},
+		{86, 0x400, 0x400},
+	}
+	for _, c := range check {
+		v, m, b := w[c.off], c.mask, c.bit
+		if (v & m) != b {
+			return fmt.Errorf("Word at offset %d: 0x%#x and should be 0x%#x", c.off, c.mask&v, b)
```

Comment:
can you make the error more descriptive for what it is? like `fmt.Errorf("not a post-2003 LBA device (your stuff too old, yo): word at offset %d ...")`